sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs

```diff
@@ -1469,16 +1496,31 @@ private static RequestContext CreateRequestContext(ErrorOptions errorOptions, Ca
 
         private class ConfigurationRequestFailedDetailsParser : RequestFailedDetailsParser
         {
+            private const string TroubleshootingText =
```

Comment:
This is too detailed, in my opinion.  We want to keep the focus on the actual error message rather than our TSG text.  This should be a generic message we tack onto the very end of the error message that just makes folks aware that it exists, like:

> For troubleshooting information, see https://aka.ms/azsdk/net/appconfiguration/troubleshoot

This is the form used for the other .NET libraries enriching errors.  Examples:

- [Event Hubs](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Resources.Designer.cs#L918)
- [Service Bus](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Azure.Messaging.ServiceBus/src/Constants.cs#L57)
- [Identity](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/src/Credentials/ManagedIdentityCredential.cs#L27)

Reply:
I agree with Jesse here. Based on similar experiences, I believe we'll often forget to update this logic if we ever update the troubleshooting guide. This is specially true for AppConfiguration where ownership rotates frequently.

Reply:
Should we have a static analyzer to check for this text field value? It may be useful in general. Also, curious if this is something we are adding to our DPG code path.

Reply:
I don't think so, and no, we're not.   Troubleshooting Guides are not generated, they are created by hand.  They often don't appear until a stable release or two.   They can't be generated upfront, because we can't predict the timing.

Reply:
I wasn't talking about generating the TSG in an automatable way. Just checking that we are consistent in TSGs like we are following generic text, aka.ms link in case it is present.

Reply:
I was talking about generating the error message enrichment.   We can't do that as part of codegen, because we can't know when the TSG is available nor the link to it.   With respect to the analyzer suggestion, I don't know how you would identify the pattern to make a generic analyzer out of it.